src/StudyEntities/Candidate/ValidatableIdentifier.php

```diff
@@ -0,0 +1,66 @@
+<?php declare(strict_types=1);
+/**
+ * Implementation of ValidatableIdentifier needs to provide a validation
+ * function to ensure that no Identifier of that type exists if the value is
+ * not validated.
+ *
+ * PHP Version 7
+ *
+ * @category StudyEntities
+ * @package  LORIS
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\StudyEntities\Candidate;
+
+/**
+ * ValidatableIdentifier with the default constructor that call validate
+ *
+ * @category StudyEntities
+ * @package  LORIS
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+abstract class ValidatableIdentifier implements Identifier
+{
+    /**
+     * The Identifier's value
+     *
+     * @var string The identifier value to be validated
+     */
+    protected $value;
+
+    /**
+     * This function is called by the contructor to ensure that no
+     * ValidatableIdentifier exists if it value is not valid.
+     *
+     * Validation should be done on the value format and not on the database
+     * state.
+     *
+     * @param string $value The value to be validated
+     *
+     * @return bool True if the value format is valid
+     */
+    abstract protected function validate(string $value): bool;
+
+    /**
+     * Default constructor
+     *
+     * This is final to make sure that classes extending ValidatableIdentifier
+     * will call validate upon instanciation.
```

Comment:
```suggestion
     * will call validate upon instantiation.
```

I'm not convinced that it needs to be declared final, but the grammar in the comment should be fixed at any rate.

Reply:
For now, I think it is ok to assume that classes extending the ValidatableIdentifier should use its constructor. But I a willing to revisit that if need be.